src/components/responsive-flyout/ResponsiveFlyout.js

```diff
@@ -0,0 +1,70 @@
+// @flow
+import * as React from 'react';
+import classNames from 'classnames';
+import { Flyout, Overlay } from '../flyout';
+import ScrollWrapper from '../scroll-wrapper';
+import type { FlyoutProps } from '../flyout/Flyout';
+import OverlayHeader from './OverlayHeader';
+
+import './styles/ResponsiveFlyout.scss';
+
+type Props = FlyoutProps & {
+    /** Components to render in the overlay */
+    children: ?React.Node,
+    /** Set className to the overlay wrapper */
+    className?: string,
+    /** Disable responsiveness behavior for higher-level components that want control */
+    enabled?: boolean,
+    /** Custom button to trigger for opening/closing the flyout */
+    flyoutButton: React.Element<any>,
+    /** Optional function to get the flyoutRef in parent components */
+    flyoutRefFn?: any => any,
+    /** Header for responsive layout */
+    header?: React.Element<any>,
+    /** Overlay classname */
+    overlayClassName?: string,
+    /** Optional function to get the scrollRef in parent components */
+    scrollRefFn?: any => any,
```

Comment:
Could we use the prop type from [`ScrollWrapper`](https://github.com/box/box-ui-elements/blob/master/src/components/scroll-wrapper/ScrollWrapper.js#L15)?

`scrollRefFn: (?HTMLElement) => void,`